enhancements/insights/config-api.md

```diff
@@ -0,0 +1,287 @@
+
+### Non-Goals
+
+- Move the majority of the current configuration options to a new configuration API - Insights has quite a lot of configuration options
+  and it does not make sense to expose everything in the new configuration API 
+- Removing the original configuration via `support` secret
+
+## Proposal
+
+The proposal is to introduce a new Insights `CustomResourceDefinition` in the OpenShift API providing a way to configure the Insights operator and to report its status. The new CRD should follow a common CRD structure providing `spec` and `status` attributes. 
+
+```go
+type InsightsSpec struct {
+	// GatheringConfig spec attribute includes all the configuration options related to
+	// gathering of the Insights archive and its uploading to the ingress.
+	GatheringConfig *GatheringConfig `json:"gatheringConfig,omitempty"`
+}
+
+type InsightsStatus struct {
+	// GatheringStatus provides basic information about the last Insights gathering.
+	GatheringStatus *GatheringStatus `json:"gatheringStatus,omitempty"`
+	// ReportStatus provides general Insights analysis results.
+	ReportStatus *ReportStatus `json:"reportStatus,omitempty"`
+}
+
+type GatheringConfig struct {
+	// EnableObfuscation is flag enabling the global obfuscation of the IP addresses and base domain
+	// in the Insights archive data.
+	EnableObfuscation bool `json:"enableObfuscation,omitempty"`
+	// ForceGatherReason enables user to force Insights data gathering by setting a new reason.
+	// When there is some gathering in the progress then it is interrupted.
+	// When all the gatherers are deactivated by the `DisabledGatherers`, nothing happens.
+	ForceGatherReason string `json:"forceGatherReason,omitempty"`
+	// List of gatherers to be excluded from the gathering. All the gatherers can disabled by providing "all" value.
+	// If all the gatherers are disabled, the Insights operator does not gather any data.
+	// The particular gatherers IDs can be found at https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md.
+	// An example of disabling some gatherers can look like `disabledGatherers: ["clusterconfig/machine_configs", "workloads/workload_info"]`
+	DisabledGatherers []string `json:"disabledGatherers,omitempty"`
+}
+
+type GatheringStatus struct {
+	// LastGatherTime is the last time when Insights gathering finished.
+	LastGatherTime metav1.Time `json:"lastGatheringTime,omitempty"`
+	// Reason provides last known reason of gathering. This is helpful
+	// especially when gathering was forced by user
+	Reason string `json:"reason,omitempty"`
+	// Status is waiting or running. If it is running then it means there is some gathering in progress.
+	// If it is waiting then a new gathering can be triggered immediately.
+	Status *GatherStatus `json:"status,omitempty"`
+	// List of active gatherers (and their statuses) in the last gathering.
+	GathererStatuses []GathererStatus `json:"gathererStatuses,omitempty"`
+}
+
+type ReportStatus struct {
```

Comment:
This also looks like a status on lower level object comparable to things like authentication.operator.openshift.io

Reply:
Duplicate. Resolving